transformer_engine/pytorch/attention.py

```diff
@@ -355,18 +369,30 @@ def forward(
         ]
 
         max_seqlen = seqlen
-        cu_seqlens = torch.arange(
-            0,
-            (batch_size + 1) * seqlen,
-            step=seqlen,
-            dtype=torch.int32,
-            device=query_layer.device)
+        if self.attn_mask_type == "padding":
+            assert (
+                attention_mask is not None
+            ), "Boolean attention mask must be provided for padding."
```

Comment:
So we're settled on requiring a mask tensor than a cu_seqlen tensor from users?

Reply:
Yeah I think so